projects/netgrif-components-core/src/lib/view/case-view/tabbed-case-view.ts

```diff
@@ -8,25 +8,34 @@ import {CaseViewService} from './service/case-view-service';
 import {SimpleFilter} from '../../filter/models/simple-filter';
 import {FilterType} from '../../filter/models/filter-type';
 import {OverflowService} from '../../header/services/overflow.service';
-import {NewCaseCreationConfigurationData} from '../../side-menu/content-components/new-case/model/new-case-injection-data';
+import {
+    NAE_NEW_CASE_CREATION_CONFIGURATION_DATA,
+    NewCaseCreationConfigurationData
+} from '../../side-menu/content-components/new-case/model/new-case-injection-data';
 import {Observable} from 'rxjs';
+import {NAE_AUTOSWITCH_TAB_TOKEN} from './models/autoswitch-token';
+import {NAE_OPEN_EXISTING_TAB} from './models/open-existing-tab-token';
 
 export interface InjectedTabbedCaseViewData extends InjectedTabData {
     tabViewComponent: Type<any>;
     tabViewOrder: number;
 }
 
-export abstract class TabbedCaseView extends AbstractCaseView {
+@Component({
+    selector: 'ncc-abstract-tabbed-case-view',
+    template: ''
+})
+export abstract class AbstractTabbedCaseViewComponent extends AbstractCaseViewComponent {
 
     private readonly _correctlyInjected;
 
     protected constructor(caseViewService: CaseViewService,
                           protected _loggerService: LoggerService,
                           @Inject(NAE_TAB_DATA) protected _injectedTabData: InjectedTabbedCaseViewData,
                           protected _overflowService?: OverflowService,
-                          protected _autoswitchToTaskTab = true,
-                          protected _openExistingTab = true,
-                          protected _newCaseCreationConfig: NewCaseCreationConfigurationData = {
+                          @Optional() @Inject(NAE_AUTOSWITCH_TAB_TOKEN) protected _autoswitchToTaskTab = true,
```

Comment:
default values specified like this don't work. The `@Optional()` annotation injects `null` if the dependency cannot be found, therefore the javascript defaults are not set since `null` is not `undefined`

Reply:
it is working as i thinks and angular forced me to do that this way, so default values works now like this (I tested it and debugged and the default value was there)